### PR TITLE
CCM-2534: Add open API spec for callback message statuses

### PR DESCRIPTION
**Pull request by TheWorldWideWebb**
## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Checklist
* [ ] Brief description of work completed, and any technical decisions made as part of the PR
* [ ] PR link added as a comment to the relevant JIRA ticket
* [ ] Branch deployed to a dynamic env - link to deployment pipeline
* [ ] PR link shared on Slack and/or Teams
* [ ] 2 reviews received
* [ ] Tester approval

### CODE REVIEW

**Review by beoh1 — DISMISSED** [review]

File: specification/documentation/callbacks/message-subscription-openapi-spec.json [hunk 1]
```diff
@@ -0,0 +1,230 @@
+{
+  "openapi": "3.1.0",
+  "info": {
+    "version": "1.0",
+    "title": "Callback Example",
+    "description": "This is an example OpenAPI specification detailing an endpoint to be created by a partner who has integrated with the Communications Manager API and wishes to use the callbacks feature. For context, please see the [Communications Manager API](https://digital.nhs.uk/developer/api-catalogue/communications-manager)."
```
Comment (adadigital) [inline specification/documentation/callbacks/message-subscription-openapi-spec.json:6]: NHS Notify*

You may notice that the URL on this line redirects to here: https://digital.nhs.uk/developer/api-catalogue/nhs-notify

Comment (TheWorldWideWebb) [reply specification/documentation/callbacks/message-subscription-openapi-spec.json]: Fixed

File: specification/documentation/callbacks/message-subscription-openapi-spec.json [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+              "channels": {
+                "type": "array",
+                "minItems": 1,
+                "items": {
+                  "$ref": "#/components/schemas/MessageStatusChannel"
+                }
+              },
+              "timestamp": {
+                "type": "string",
+                "description": "Timestamp of when the callback was generated.",
```
Comment (Ian-Hodges) [inline specification/documentation/callbacks/message-subscription-openapi-spec.json:117]: This should be the timestamp of when the dynamodb record was inserted/updated.

Comment (TheWorldWideWebb) [reply specification/documentation/callbacks/message-subscription-openapi-spec.json:117]: Its the same thing, the time at which the message was created/inserted is also the time this callback would be generated. I would say its more user friendly to keep it like this to allow consumers to order their callbacks which we don't support.

Comment (Ian-Hodges) [reply specification/documentation/callbacks/message-subscription-openapi-spec.json:117]: They are roughly equal, but only when things are running smoothly. If I was receiving these messages I would want to know the timestamp from dynamodb not when the callback request was created.
It's a small change which I've squeezed CCM-3548.

**Review by adadigital — APPROVED** [review]

**Review by beoh1 — APPROVED** [review]

**Pull request merged**